community/io/src/main/java/org/neo4j/io/pagecache/PageSwapperFactory.java

```diff
@@ -33,17 +34,27 @@
  * <p>
  * The PageSwapperFactory presumably knows about what file system to use.
  * <p>
+ * To be able to create PageSwapper factory need to be configured first using appropriate configure call.
  * Note that this API is <em>only</em> intended to be used by a {@link PageCache} implementation.
  * It should never be used directly by user code.
  */
 public interface PageSwapperFactory
 {
     /**
-     * Configure the FileSystemAbstraction to use.
-     * <p>
-     * This must be called before the first PageSwapper is created.
+     * Configure page swapper factory with filesystem and config
+     * @param fs file system to use in page swappers
+     * @param config custom page swapper configuration
+     */
+    void configure( FileSystemAbstraction fs, Configuration config );
+
+    /**
+     * Configure swapper with filesystem to use.
+     * @param fs file system to use in page swappers
      */
-    void setFileSystemAbstraction( FileSystemAbstraction fs );
+    default void configure( FileSystemAbstraction fs )
+    {
+        this.configure( fs, Configuration.EMPTY );
+    }
```

Comment:
I think we can get away with only having one `configure` method; the one that takes both parameters.
I'd also like to call it `configureAndInitialise`, to highlight that it has important life-cycle semantics. Or alternatively, we can call it `open`, to be symmetric with the `close` method further down.